### PR TITLE
Fix: Explicit validation of custom field name unique constraint

### LINKED ISSUE #5643: [BUG] Adding a custom field with the same name of an existing field leads to internal server error
### Description

If you try to add a custom field with the name of an already existing custom field name, an internal error occours.

### Steps to reproduce

1. Create custom field "ABC" with type integer
1. Create another custom field "ABC" with type string

### Webserver logs

```bash
{"headers":{"normalizedNames":{},"lazyUpdate":null},"status":500,"statusText":"Internal Server Error","url":"http://paperless.local/api/custom_fields/","ok":false,"name":"HttpErrorResponse","message":"Http failure response for http://paperless.local/api/custom_fields/: 500 Internal Server Error","error":"\n<!doctype html>\n<html lang=\"en\">\n<head>\n  <title>Server Error (500)</title>\n</head>\n<body>\n  <h1>Server Error (500)</h1><p></p>\n</body>\n</html>\n"}
```

### Browser logs

_No response_

### Paperless-ngx version

2.4.2

### Host OS

GNU/Linux synology_braswell_216

### Installation method

Docker - official image

### Browser

Firefox

### Configuration changes

_No response_

### Other

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

Closes #5643

## Type of change

- [x] Bug fix: non-breaking change which fixes an issue.
- [ ] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: src/documents/serialisers.py [hunk 1]
```diff
@@ -441,6 +441,17 @@ class Meta:
             "data_type",
         ]
 
+    def validate(self, attrs):
+        # see https://github.com/encode/django-rest-framework/issues/7173
```
Comment (stumpylog) [inline src/documents/serialisers.py:445]: Could you make this a TODO maybe, so it's easier to find when (if) 3.15 is released?

Comment (shamoon) [reply src/documents/serialisers.py]: Sure yea! Took the liberty of changing the other one too

**Review by stumpylog — APPROVED** [review]
Looks great.  I was going to be lazy and not even add a test in fix/5643-unique

**Comment by shamoon** [comment]
Ah sorry to duplicate the work, thought you hadn’t gotten to it yet.

Yea I fear the codecov ❌

**Pull request merged**